risc0/povw/guests/log-builder/src/main.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2025 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::Context;
+use risc0_povw::{
+    guest::{Input, Journal, State},
+    Job, WorkLog,
+};
+use risc0_zkvm::{guest::env, sha::Digestible, Digest};
+use ruint::Uint;
+
+type U96 = Uint<96, 2>;
+
+fn main() {
+    let input: Input = borsh::from_slice(&env::read_frame()).unwrap();
```

Comment:
Why borsh?

Reply:
It's not a strong preference. I considered `bincode`, `postcard` and `borsh` and ended up deciding to use `borsh` because it guarantees the property of its encodings being canonical, which prevents issues with `env::verify` [below](https://github.com/risc0/risc0/blob/25c5694a2daae335b09f65a1a98b2a684c324352/risc0/povw/guests/log-builder/src/main.rs#L31) in that if when re-serialized the journal does not match what was committed, the verification will fail.